cli/version_check.go

```diff
@@ -13,7 +13,7 @@ import (
 
 // The terraform --version output is of the format: Terraform v0.9.5-dev (cad024a5fe131a546936674ef85445215bbc4226+CHANGES)
 // where -dev and (commitid+CHANGES) is for custom builds or if TF_LOG is set for debug purposes
-var TERRAFORM_VERSION_REGEX = regexp.MustCompile("Terraform (v?[\\d\\.]+)(?:-dev)?(?: .+)?")
+var TERRAFORM_VERSION_REGEX = regexp.MustCompile("Terraform (v?[\\d\\.]+)(?:-dev)?(?:-beta\\d+)?(?: .+)?")
```

Comment:
Is it worth matching exactly what comes after the dash? Before it was `-dev`; today it's `-beta`; in the future, it might be something else. It seems like we could make this slightly more maintainable by ignoring what comes after the dash, right?

Reply:
That's a good point, I'll update the PR later this day.

Reply:
https://gist.github.com/artemsablin/d56a2b38421751374782814dddba9b36